### PR TITLE
Frontend behind path based proxy

### LINKED ISSUE #56: [FEAT] Frontend behind path based proxy
**Is your feature request related to a problem? Please describe.**
I run scrutiny behind path based nginx proxy
```
    location ~ ^/scrutiny(/?)(.*) {
         proxy_pass http://127.0.0.1:45000/$2;
    }
```

Once I load https://mydomain.com/scrutiny the page send a 302 with `Location: /web`, and then after the redirection the location directive in nginx doesn't match anymore

**Describe the solution you'd like**
Consider using a base path config for frontend (if possible at all) or build the frontend urls from current url

**Additional context**
Scrutiny doesn't provide any https listener, so I need to proxy it by nginx

### PR DESCRIPTION
**Pull request by francescayeye**
Fix #56

### CODE REVIEW

**Comment by francescayeye** [comment]
few refinement needed

**Comment by AnalogJ** [comment]
Hey @aspacca 
Thanks so much for this PR! 🎉 

Can you add some tests for the `basepath` work that you've done? I just want to make sure that everything continues to function the same way for existing users.

**Comment by AnalogJ** [comment]
lol, I just saw that you made a commit tweaking the existing tests 😄 
I'll wait to comment until you're finished.

Thanks again!

**Comment by francescayeye** [comment]
@AnalogJ I'm done with the tests, but I don't see any checks executed in github, what's wrong?

**Comment by AnalogJ** [comment]
Yeah, it looks like I misconfigured the gihtub action. Can you push an empty commit to your PR. Or close this PR and open a new one (with the same branch)? That should trigger the updated Github action.

File: webapp/backend/pkg/web/server.go [hunk 1]
```diff
@@ -27,28 +27,33 @@ func (ae *AppEngine) Setup(logger logrus.FieldLogger) *gin.Engine {
 	r.Use(middleware.ConfigMiddleware(ae.Config))
 	r.Use(gin.Recovery())
 
-	api := r.Group("/api")
+	basePath := ae.Config.GetString("web.src.backend.basepath")
```
Comment (AnalogJ) [inline webapp/backend/pkg/web/server.go:30]: can you explicitly set a default value (`""`) in [Config class](https://github.com/AnalogJ/scrutiny/blob/master/webapp/backend/pkg/config/config.go)

Also, this should be added to the [`example.scruiny.yaml`](https://github.com/AnalogJ/scrutiny/blob/master/webapp/backend/pkg/config/config.go)

**Comment by AnalogJ** [comment]
Hey @aspacca this is looking good.

The last thing I want to check is if this will continue to work with the local development (non-docker) environment: 

https://github.com/AnalogJ/scrutiny/blob/master/CONTRIBUTING.md#frontend

**Comment by francescayeye** [comment]
> The last thing I want to check is if this will continue to work with the local development (non-docker) environment:

I will give it a try, for sure you don't have to pass `--deploy-url="/web/" --base-href="/web/"`

**Comment by francescayeye** [comment]
done, it was only a matter of chaning the instructions

**Comment by flying-sausages** [comment]
did the contrib file edit in 9405ec2 break the arm build? hwat?

**Comment by AnalogJ** [comment]
@flying-sausages no, I changed the CI/Release scripts. The ARM build breakage was all me.

**Comment by francescayeye** [comment]
@AnalogJ any news on when this can be merged?

File: example.scrutiny.yaml [hunk 2]
```diff
@@ -26,6 +26,9 @@ web:
   src:
     frontend:
       path: /scrutiny/web
```
Comment (flying-sausages) [inline example.scrutiny.yaml]: Could it be a good idea to rename this to distinguish `path` from `basepath`? Might be better to change it now than delay this for down the road.

**Comment by flying-sausages** [comment]
beep boop @AnalogJ hiya

**Comment by AnalogJ** [comment]
apologies @flying-sausages  I've been distracted with some personal stuff. I'll take another look at this PR this weekend.

**Comment by francescayeye** [comment]
@AnalogJ any news?

**Comment by francescayeye** [comment]
@AnalogJ, is the project still active? :)

**Comment by AnalogJ** [comment]
Hey, sorry, I'm working on some other personal stuff right now, but I expect to pick up `scrutiny` again within the month.

**Comment by francescayeye** [comment]
@AnalogJ any news?

**Comment by NemesisRE** [comment]
would love to use basePath for this project as I have everything proxied by path

**Comment by mcesnik** [comment]
I did notice an "error" in my logs although I could not tell if this was actually a problem...
```
time="2021-09-14T14:04:00-04:00" level=info msg="127.0.0.1 - 1c284a96e25f [14/Sep/2021:14:04:00 -0400] \"POST /api/devices/register\" 200 5784 \"\" \"Go-http-client/1.1\" (1ms)" clientIP=127.0.0.1 hostname=1c284a96e25f latency=1 method=POST path=/api/devices/register referer= respLength=5784 statusCode=200 userAgent=Go-http-client/1.1

2021/09/14 14:04:00 ERROR: invalid character '<' looking for beginning of value
```

The path here was not updated correctly to use the `basepath` (in my case `/scrutiny/`).

**Side Note** @AnalogJ I noticed in this case the server returns 200 OK even though it should probably have returned a 404 NotFound

**Comment by francescayeye** [comment]
@mcesnik 
I didn't get what was the error exactly

**Comment by cardboardpig** [comment]
Like other commenters I would love to be able to set the path! I saw that `example.scrutiny.yaml` contained a `web.src.frontend.path` attribute and figured it was possible but after it did not work I found this issue.

**Pull request closed without merging**